router/client/src/lib.rs

```diff
@@ -44,3 +47,34 @@ impl From<transport::Error> for ClientError {
 }
 
 pub type Result<T> = std::result::Result<T, ClientError>;
+
+// Small convenience re-wrapping of `Chunk`.
+impl From<Chunk> for InputChunk {
+    fn from(chunk: Chunk) -> Self {
+        InputChunk { chunk: Some(chunk) }
+    }
+}
+
+/// Convert input chunks to a stringly-typed input for backwards
+/// compat for backends that haven't implemented chunked inputs.
+pub trait ChunksToString {
+    /// Convert chunks to string.
+    fn chunks_to_string(&self) -> String;
+}
+
+impl ChunksToString for Vec<InputChunk> {
+    fn chunks_to_string(&self) -> String {
+        let mut output = String::new();
+        self.iter().for_each(|c| match &c.chunk {
+            Some(Chunk::Text(text)) => output.push_str(text),
+            Some(Chunk::Image(Image { data, mimetype })) => {
+                let encoded = STANDARD.encode(data);
+                output.push_str(&format!("![](data:{};base64,{})", mimetype, encoded))
+            }
+            // We don't create empty chunks, probably better to be robust
+            // than unreachable!().
+            None => {}
```

Comment:
Just wanted to make sure you see this. Since we are creating the chunks in the router, this case seems to never occur. But we could also have `unreachable!()` or bubble up an error. I don't have a strong opinion.

Reply:
Either is fine.
Seems to be an artifact of the protobuf definition.

- `unreachable` makes thing hard crash so easier to debug, but less robust
- do nothing, harder to debug should anything go wrong (if even visible) but more robust.

I don't a strong opinion to be honest. Insinctively I'd make it a `unreachable` or equivalent, at the very least in the silent path I would add a warn log.